### PR TITLE
Add STAmount calc switchover support to tx queue

### PR DESCRIPTION
**Pull request by ximinez**
These commits will get squashed before merging, so no need to individually review. I just made some last minute tweaks and want to confirm CI still passes.

### CODE REVIEW

File: src/test/app/Flow_test.cpp [hunk 1]
```diff
@@ -1282,10 +1282,19 @@ struct Flow_test : public beast::unit_test::suite
             testSelfPayLowQualityOffer({fs...});
         };
         testWithFeats();
+        // Don't bother testing the old payment engine with the queue,
+        // because it's old and buggy. That's why we updated it.
+        //testWithFeats(featureFeeEscalation);
```
Comment (seelabs) [inline src/test/app/Flow_test.cpp]: I'd just remove it all together rather than leave the dead code commented out.

**Comment by seelabs** [comment]
This is transaction breaking and needs an amendment.

File: src/test/app/Flow_test.cpp (hunk 1 above)
Comment (ximinez) [inline src/test/app/Flow_test.cpp]: Fixed.

**Comment by ximinez** [comment]
TODO: change the std::unique_ptr in that last commit to boost::optional

**Comment by ximinez** [comment]
The new functionality in `TxQ` is now behind an amendment. I also updated the `Offer` and `Flow` tests so they can run with (in principal) any combo of amendments, but updated to only run with the most realistic combo (fee escalation and the new fix1513). Finally, I added `Offer_manual` and `Flow_manual` manual tests with all the relevant combos of amendments because they take way too long to run every time.

**Comment by ximinez** [comment]
And rebased again up to rc1.

File: src/test/app/Flow_test.cpp [hunk 2]
```diff
@@ -1254,6 +1254,34 @@ struct Flow_test : public beast::unit_test::suite
             ter(temBAD_PATH));
     }
 
+    template<class... T>
+    void testWithFeats(bool haveFlow, T&&... fs)
```
Comment (seelabs) [inline src/test/app/Flow_test.cpp]: We can get rid of the `bool` parameter and check for `haveBool` directly in the param pack, but it requires the changes in this PR: https://github.com/ripple/rippled/pull/2216

I'd like to make that change, but I think the easiest is to wait for both PRs to be merged and to created another PR later with the change.

File: src/test/app/Flow_test.cpp [hunk 3]
```diff
@@ -1262,35 +1290,50 @@ struct Flow_test : public beast::unit_test::suite
...
+            featureFeeEscalation, fix1513);
+        testEmptyStrand({featureFlow, fix1373, featureFlowCross,
+            featureFeeEscalation, fix1513 });
+    }
+};
+
+struct Flow_manual_test : public Flow_test
+{
+    void run() override
+    {
```
Comment (seelabs) [inline src/test/app/Flow_test.cpp]: I like this change a lot. Good thinking!

File: src/test/app/Flow_test.cpp [hunk 4]
```diff
@@ -1254,6 +1254,34 @@ struct Flow_test : public beast::unit_test::suite
...
+        testUnfundedOffer(true, { fs... });
+        testUnfundedOffer(false, { fs... });
+        testReexecuteDirectStep({ fix1368, fs... });
+        testSelfPayLowQualityOffer({ fs... });
+    }
+
+    template<class... T>
+    void testWithFeats(T&&... fs)
+    {
+        testWithFeats(!!sizeof...(fs), fs...);
```
Comment (seelabs) [inline src/test/app/Flow_test.cpp]: I prefer `sizeof...(fs) > 0` or `sizeof...(fs) != 0` to `!!`. But that's clearly just a personal preference, no change needed. Plus this code will go away later anyway.

Comment (ximinez) [reply src/test/app/Flow_test.cpp]: I'll leave it for later, then.

File: src/test/app/Flow_test.cpp (hunk 3 above)
Comment (ximinez) [reply src/test/app/Flow_test.cpp]: Thanks!

**Comment by ximinez** [comment]
* Rebased on 0.80.0-rc2
* Resolved conflicts
* Squashed
* Updated the commit message (@seelabs @miguelportilla Could you check the commit message out?)

**Comment by seelabs** [comment]
Commit message looks good to me. If I were to nit, I'd probably call it `STAmount switchover` and leave out the `calc`, but it really doesn't matter. Fine as is.

**Comment by ximinez** [comment]
I'll change it since I realized I forgot the `:`

**Comment by seelabs** [comment]
In 0.90.0-b1

**Pull request closed without merging**